bitvector8.go

```diff
@@ -91,3 +91,62 @@ func (b Bitvector8) BitIndices() []int {
 
 	return indices
 }
+
+// Contains returns true if the bitlist contains all of the bits from the provided argument
+// bitlist. This method will panic if bitlists are not the same length.
+func (b Bitvector8) Contains(c Bitvector8) bool {
+	if b.Len() != c.Len() {
+		panic("bitvector are different lengths")
+	}
+
+	// To ensure all of the bits in c are present in b, we iterate over every byte, combine
+	// the byte from b and c, then XOR them against b. If the result of this is non-zero, then we
+	// are assured that a byte in c had bits not present in b.
+	for i := 0; i < len(b); i++ {
+		if b[i]^(b[i]|c[i]) != 0 {
+			return false
+		}
+	}
+
+	return true
+}
+
+// Overlaps returns true if the bitlist contains one of the bits from the provided argument
+// bitlist. This method will panic if bitlists are not the same length.
+func (b Bitvector8) Overlaps(c Bitvector8) bool {
+	lenB, lenC := b.Len(), c.Len()
+	if lenB != lenC {
+		panic("bitlists are different lengths")
+	}
+
+	if lenB == 0 || lenC == 0 {
+		return false
+	}
+
+	// To ensure all of the bits in c are not overlapped in b, we iterate over every byte, invert b
+	// and xor the byte from b and c, then and it against c. If the result is non-zero, then
+	// we can be assured that byte in c had bits not overlapped in b.
+	for i := 0; i < len(b); i++ {
+		// If this byte is the last byte in the array, mask the length bit.
+		mask := uint8(0xFF)
+
+		if (^b[i]^c[i])&c[i]&mask != 0 {
+			return true
+		}
+	}
+	return false
+}
+
+// Or returns the OR result of the two bitfields. This method will panic if the bitlists are not the same length.
+func (b Bitvector8) Or(c Bitvector8) Bitvector8 {
+	if b.Len() != c.Len() {
+		panic("bitlists are different lengths")
+	}
+
+	ret := make([]byte, len(b))
+	for i := 0; i < len(b); i++ {
+		ret[i] = b[i] | c[i]
+	}
+
+	return ret
```

Comment:
Same here, no loop required.
```suggestion
	return b[0] | c[0]
```